### DISCUSSION TITLE
Reviews of pull requests are stalled

### DISCUSSION THREAD

**Discussion by bernhardkaindl** [spack/spack#25759] [Q&A] [answered]
Hi,

- Reviews of pull requests are stalled,
- pull requests pile up and nobody reviews or merges them.

What's up? When are merges/reviews of pull requests being resumed?

**Comment by tgamblin** [comment, marked as answer]
I’m not sure the evidence supports this statement. We’ve merged 85 pull requests in the past week and 372 in the past month:

https://github.com/spack/spack/pulse

The length of the queue is not indicative of the quality of service.

If there’s a particular pull request you’d like reviewed feel free to link it here or ping maintainers about it.

**Reply by haampie** [reply]
To be fair, getting PRs merged for spack internals takes exceptionally long compared to other open source projects I have contributed to. Most of the time the reason is lack of review.

One of my PRs fixes a security issue, and it's open for exactly 3 months now with regular rebases from my side and requests to get it reviewed/merged.

**Reply by tgamblin** [reply]
We’ve merged 60 of your PRs in the past 30 days (16% of all PRs), a number of which are core PRs. We appreciate all of them!

The security issue you mentioned is not a critical one — and it’s not so much with spack as with users putting secrets in their environment. To be fair, it’s a good PR and we’re going to merge it.

Not saying we couldn’t review things faster, but the volume of PRs we get is very high and we still manage to do a lot of core feature development and merge many of them. I think everyone feels pretty swamped. Right now we are trying to replace a core feature of spack (the concretizer) without affecting users, introducing binary bootstrapping to the mainline for the first time, and preparing for the 0.17 release. We’re also trying to help ECP users prepare for three new architectures at the same time.

I suspect things will speed up after the release and after 2021 milestones are done (end of September), but please bear with us until then. 

Thanks for all your help with spack!
